### ISSUE TITLE
Feature Request: Enable HEAD requests / return Header info only

### ISSUE THREAD

**Issue by 01dr1ch** [MISP/MISP#9129]
### Is your feature request related to a problem? Please describe.

When filtering Attributes (restSearch or PyMISP), there is currently no option to let API call return _only_ information about count of Attributes matching the filter (X-Result-Count header in REST API).

There is this option in GUI Rest Client: "Show results" that prevents from loading the data into GUI. No similar option is available through neither API (REST/PyMISP).

Moreover, checking the inner code, "Show Results" checkbox lets MISP process the query anyway (sometimes __tens of thousands__ of Attributes - utilizes MISP), have the results ready and then it just drops the body with results in case "Show Results" checkbox is unchecked.

Code excerpt from ```/var/www/MISP/app/Controller/ApiController.php```

``` 
  if (!empty($request['show_result'])) {
      $viewData['data'] = $response->body;
  } else {
      if ($response->isOk()) {
          $viewData['data'] = 'Success.';
      } else {
          $viewData['data'] = 'Something went wrong.';
      }
  }
```

In other words MISP has no option to acquire only COUNT of matching Attributes. Loading them and then dropping off results is obviously suboptimal.

### Describe the solution you'd like

Suggesting to add simple FLAG into the PyMISP and REST API calls that would indicate only Headers are requested, not the hard data themselves.

### Describe alternatives you've considered

Setting "limit": "1" doesn't work, since X-Result-Count equals the value of limit. If I have 2000 matching Attributes and limit the results to show only 1, X-Result-Count equals 1 as well. This is by design as I remember.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct
